core/commands/pin/remotepin.go

```diff
@@ -240,6 +260,22 @@ NOTE: a comma-separated notation is supported in CLI for convenience:
 	},
 }
 
+func isOriginAddr(addr ma.Multiaddr, noAnnounceAddrs []string) bool {
+	// IP is located at the second index: /ip4/127.0.0.1/tcp/8080
+	ip := net.ParseIP(strings.Split(addr.String(), "/")[2])
+	if ip.IsLoopback() {
+		return false
+	}
+
+	for _, noAnnounceAddr := range noAnnounceAddrs {
+		if strings.HasPrefix(addr.String(), noAnnounceAddr) {
+			return false
+		}
+	}
```

Comment:
Turns out we don't need to do `NoAnnounce` filtering here, it is already applied to `node.PeerHost`, so nothing to do here :)

```suggestion

```

Sidenote:

If we needed this, this would not  work for  all cases, because `NoAnnouce`  are usually definition of specific network submask, the last  segment after `/ipcidr`  being the size of the network – see examples [here](https://github.com/ipfs/go-ipfs-config/blob/0a474258a95d8d9436a49539ad9d7da357b015ab/profile.go#L24-L43)

For futre reference, the  proper way of checking if maddr is blocked is already implemented in [core/node/libp2p/addrs.go](https://github.com/ipfs/go-ipfs/blob/v0.9.0-rc1/core/node/libp2p/addrs.go) in form of `ma.NewFilters` and  `filters.AddrBlocked`